File: backend/app/schemas/understanding.py
import uuid
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, ConfigDict, Field


class ProjectUnderstandingResponse(BaseModel):
    """Deterministic structured representation of project understanding across 11 dimensions."""
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    status: str
    problem: Optional[str] = None
    target_users: List[str] = Field(default_factory=list)
    objectives: List[str] = Field(default_factory=list)
    requirements_summary: Optional[str] = None
    modules: List[str] = Field(default_factory=list)
    tech_stack: List[str] = Field(default_factory=list)
    architecture_overview: Optional[str] = None
    dependencies: List[str] = Field(default_factory=list)
    expected_scale: Optional[str] = None
    deployment: Optional[str] = None
    team: List[Dict[str, Any]] = Field(default_factory=list)
    provenance: Dict[str, Any] = Field(default_factory=dict)
    source_artifact_ids: List[str] = Field(default_factory=list)
    extracted_sections_count: int = 0
    total_words_analyzed: int = 0
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/understanding.ts
export interface TeamMember {
  name: string;
  role?: string | null;
}

export interface ProvenanceRecord {
  source_type: "project_metadata" | "artifact";
  field?: string;
  artifact_id?: string;
  section?: string;
  value?: string;
}

export interface ProjectUnderstanding {
  id: string;
  project_id: string;
  status: "pending" | "completed" | "failed";
  problem?: string | null;
  target_users: string[];
  objectives: string[];
  requirements_summary?: string | null;
  modules: string[];
  tech_stack: string[];
  architecture_overview?: string | null;
  dependencies: string[];
  expected_scale?: string | null;
  deployment?: string | null;
  team: TeamMember[];
  provenance: Record<string, any>;
  source_artifact_ids: string[];
  extracted_sections_count: number;
  total_words_analyzed: number;
  created_at: string;
  updated_at: string;
}
